app/app.rb

```diff
@@ -50,12 +52,19 @@ class App < Padrino::Application
     end
 
     get :index, :with => :endpoint do
-      content_type :json
+      endpoint = params.delete('endpoint')
+
+      format = File.extname(endpoint)
+      endpoint = File.basename(endpoint, format)
+      format[0] = ''
+      format = 'json' if format.empty?
+      content_type format.to_sym
```

Comment:
Is there any security risk of allowing the requestor to specify the resulting content_type?